packages/ember-glimmer/tests/integration/application/engine-test.js

```diff
@@ -572,4 +587,41 @@ moduleFor('Application test: engine rendering', class extends ApplicationTest {
       });
     });
   }
+
+  ['@test query params don\'t have stickiness by default between model'](assert) {
+    assert.expect(1);
+    let tmpl = '{{#link-to "blog.category" 1337}}Category 1337{{/link-to}}';
+    this.setupAppAndRoutableEngine();
+    this.additionalEngineRegistrations(function() {
+      this.register('template:category', compile(tmpl));
+    });
+
+    return this.visit('/blog/category/1?type=news').then(() => {
+      let suffix = '/blog/category/1337';
+      let href = this.element.querySelector('a').href;
+
+      // check if link ends with the suffix
+      assert.ok(this.stringsEndWith(href, suffix));
+    });
+  }
+
+  ['@test query params in customized controllerName have stickiness by default between model'](assert) {
```

Comment:
I stand corrected - this change doesn't break custom `controllerName` case. There was an error in the way i register the route causing the test to fail. I've updated the setup and now it's passing.